components/authz-service/server/v2/authz.go

```diff
@@ -211,19 +211,34 @@ func (s *authzServer) ValidateProjectAssignment(
 		return &api.ValidateProjectAssignmentResp{}, nil
 	}
 
-	err := s.store.EnsureNoProjectsMissing(ctx, req.ProjectIds)
-	if err != nil {
-		if _, ok := err.(*projectassignment.ProjectsMissingError); ok {
-			return nil, status.Error(codes.NotFound, err.Error())
+	newProjects := req.NewProjects
+	if len(newProjects) == 0 {
+		newProjects = []string{}
+	}
+	oldProjects := req.OldProjects
+	if len(oldProjects) == 0 {
+		oldProjects = []string{}
+	}
```

Comment:
Are these checking for the nil case? Is that possible to pass in with our validation? I feel like we should prevent this from being possible so that the API doesn't get used incorrectly by accident.

Reply:
Yeah we should verify 👍

Reply:
No way to differentiate []string{} :( . We'll have to continue assuming that a nil projects field means (unassigned) (which seems ok to me)